userapi/internal/api.go

```diff
@@ -414,3 +417,42 @@ func (a *UserInternalAPI) PerformAccountDeactivation(ctx context.Context, req *a
 	res.AccountDeactivated = err == nil
 	return err
 }
+
+// PerformOpenIDTokenCreation creates a new token that a relying party uses to authenticate a user
+func (a *UserInternalAPI) PerformOpenIDTokenCreation(ctx context.Context, req *api.PerformOpenIDTokenCreationRequest, res *api.PerformOpenIDTokenCreationResponse) error {
+	token := util.RandomString(24)
+	tokenHash := getTokenHash(token)
+	nowMS := time.Now().UnixNano() / int64(time.Millisecond)
+	expiresMS := nowMS + (3600 * 1000) // 60 minutes
```

Comment:
Given we don't use `nowMS` anymore, why not just do `time.Now().Add(60 * time.Minute).UnixNano() / int64(time.Millisecond)` as `expiresMS`?

Reply:
Also, the spec doesn't seem to mention an expiration time, so this should probably be configurable in the YAML.

Reply:
Definitely no expiry in the spec, making this part of the config makes sense. Added to the user API config.